### DISCUSSION TITLE
Disaster Recovery configuration error

### DISCUSSION THREAD

**Discussion by robv8r** [datalust/seq-tickets#1956] [Help]
When I attempt to set up [Disaster Recovery per the documentation](https://docs.datalust.co/docs/setting-up-a-disaster-recovery-dr-instance#getting-started), I receive the following error:

```
[19:50:57 INF] Moving the internal metastore to PostgreSQL
[19:50:58 INF] Opening metastore /data/Documents/metastore.flare
[19:50:58 INF] Closing metastore
[19:50:58 ERR] Metastore reconfiguration failed
Flare.Ffi.Result.FlareException: Flare native storage failed (IOError), error attempting to lock the store.
   caused by: could not acquire an exclusive lock on: "/data/Documents/metastore.flare"
   caused by: I/O error at path: "/data/Documents/metastore.flare"
   caused by: Resource temporarily unavailable (os error 11)
   at Flare.Ffi.Result.FlareResult.EnsureSuccess()
   at Flare.Ffi.Result.FlareResult.HandleNonSuccessWith(NonSuccessResultHandling handling)
   at Flare.Ffi.Bindings.StoreOpen(IntPtr path, UIntPtr pathLen, IntPtr limits, StoreHandle& store, NonSuccessResultHandling nonSuccessResultHandling)
   at Flare.Store.OpenStandalone(String path, StoreLimits limits)
   at Quince.Storage.Native.NativeDocumentStore..ctor(String path)
   at Seq.Server.StorageSubsystem.ConstructInnerMetastore(String documentsPath, SeqConfig config)
   at Seq.Cli.Shared.OfflineDocumentStore.CreateDefault(StorageLayout layout, SeqConfig config)
   at Seq.Cli.Metastore.ToPostgreSqlCommand.Run(TextWriter cout)
```

I'm using the following command to configure DR

```
seqsvr metastore to-postgresql --connection-string="Host=seq-db.logging.svc:5432;Database=seq;Username=seq;Password=P@ssw0rd"
```

Seq is deployed and functioning on Rancher (Kubernetes) v1.25.12 using version 2023.4.0 of the official Helm chart.

**Comment by liammclennan** [comment]
Hi @robv8r 

The migration is unable to lock the metastore, which suggests that Seq is running, or another Seq process is using the same data directory.

**Reply by robv8r** [reply]
How do I stop Seq in a Kubernetes Pod?

**Reply by liammclennan** [reply]
I don't know much about Kubernetes but I expect you stop or remove the pod and use an [init script](https://docs.datalust.co/docs/init-scripts) to do something before Seq starts. That will work so long as there are no other Seq processes accessing the same data directory.

**Reply by robv8r** [reply]
Is there documentation regarding Disaster Recovery, High Availability, and Scale Out for Seq in Kubernetes?

**Reply by liammclennan** [reply]
Seq does not yet support high availability or scale out. 

Here is the Disaster Recovery documentation https://docs.datalust.co/docs/setting-up-a-disaster-recovery-dr-instance. It is not specific to Kubernetes.

**Reply by robv8r** [reply]
Thank you for the information, and the link.

The [official Helm chart](https://docs.datalust.co/docs/using-helm) includes a single Pod Deployment.  I didn't see a way to mount additional volumes for things like init scripts.  I also didn't see a way to create a Follower Pod that shared the same Ingress.

How do you recommend we apply the Disaster Recovery instructions to Kubernetes using the official Helm Chart?

**Reply by KodrAus** [reply]
Hi @robv8r :wave:

The Helm chart is currently geared towards simple single-node Seq deployments. For more advanced multi-node deployments I think your best bet would be to take the chart as a starting point and assemble your own deployment that's best suited for your Kubernetes environment. There's nothing Kubernetes-specific in how Seq does multi-node deployments right now, but if you have any questions on the Seq side of things, and whether you're adding DR to existing instance or a new one, please feel free to reach out to us at `support@datalust.co` and we can run through any specifics for your environment.

**Reply by KodrAus** [reply]
>  I didn't see a way to mount additional volumes for things like init scripts.

As of the `2023.4.1` release of the chart you can specify `extraEnvs`, `extraVolumes`, and `extraVolumeMounts` that should make this a bit easier to do.
